### ISSUE TITLE
ng test crashes

### ISSUE THREAD

**Issue by dadepo** [angular/angular-cli#13027]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [x] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

macOS Mojave

node --version
v8.11.3

npm --version
5.6.0

ng --version
Angular CLI: 7.0.5
Node: 8.11.3
OS: darwin x64
Angular: 7.0.3
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router, upgrade

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.10.5
@angular-devkit/build-angular     0.10.5
@angular-devkit/build-optimizer   0.10.5
@angular-devkit/build-webpack     0.10.5
@angular-devkit/core              7.0.5
@angular-devkit/schematics        7.0.5
@angular/cli                      7.0.5
@ngtools/webpack                  7.0.5
@schematics/angular               7.0.5
@schematics/update                0.10.5
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.19.1

### Repro steps

Running `ng test` runs the test, but when the tests ends, instead of existing, it crashes.

### The log given by the failure

The stack trace when it crashes is:

```
[31m21 11 2018 20:48:44.267:ERROR [karma]: [39mTypeError: Cannot read property 'on' of undefined
    at buildProcessTree (/Users/me/ui/node_modules/tree-kill/index.js:83:15)
    at /Users/me/ui/node_modules/tree-kill/index.js:104:11
    at Array.forEach (<anonymous>)
    at ChildProcess.onClose (/Users/me/ui/node_modules/tree-kill/index.js:99:31)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Socket.stream.socket.on (internal/child_process.js:346:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:557:12)
[36m21 11 2018 20:48:44.270:DEBUG [launcher]: [39mDisconnecting all browsers
[31m21 11 2018 20:48:44.271:ERROR [karma]: [39mTypeError: Cannot read property 'on' of undefined
    at buildProcessTree (/Users/me/ui/node_modules/tree-kill/index.js:83:15)
    at /Users/me/ui/node_modules/tree-kill/index.js:104:11
    at Array.forEach (<anonymous>)
    at ChildProcess.onClose (/Users/me/ui/node_modules/tree-kill/index.js:99:31)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Socket.stream.socket.on (internal/child_process.js:346:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:557:12)

....

[36m21 11 2018 20:48:49.598:DEBUG [launcher]: [39mDisconnecting all browsers
[31m21 11 2018 20:48:49.598:ERROR [karma]: [39m{ Error: spawn pgrep EAGAIN
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  code: 'EAGAIN',
  errno: 'EAGAIN',
  syscall: 'spawn pgrep',
  path: 'pgrep',
  spawnargs: [ '-P', 2668 ] }
[36m21 11 2018 20:48:49.598:DEBUG [launcher]: [39mDisconnecting all browsers
[31m21 11 2018 20:48:49.598:ERROR [karma]: [39m{ Error: spawn pgrep EAGAIN
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  code: 'EAGAIN',
  errno: 'EAGAIN',
  syscall: 'spawn pgrep',
  path: 'pgrep',
  spawnargs: [ '-P', 2668 ] }
[36m21 11 2018 20:48:49.598:DEBUG [launcher]: [39mDisconnecting all browsers
[31m21 11 2018 20:48:49.599:ERROR [karma]: [39m{ Error: spawn pgrep EAGAIN
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  code: 'EAGAIN',
  errno: 'EAGAIN',
  syscall: 'spawn pgrep',
  path: 'pgrep',
  spawnargs: [ '-P', 2668 ] }
[36m21 11 2018 20:48:49.599:DEBUG [launcher]: [39mDisconnecting all browsers
[31m21 11 2018 20:48:49.599:ERROR [karma]: [39m{ Error: spawn pgrep EAGAIN
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  code: 'EAGAIN',
  errno: 'EAGAIN',
  syscall: 'spawn pgrep',
  path: 'pgrep',
  spawnargs: [ '-P', 2668 ] }
[36m21 11 2018 20:48:49.599:DEBUG [launcher]: [39mDisconnecting all browsers
[36m21 11 2018 20:48:49.907:DEBUG [launcher]: [39mDisconnecting all browsers

```

### Desired functionality

Expected behaviour is that tests should run, end, exit and not crash.

### Mention any other details that might be useful

The tests is being run with Chrome headless. Using `"karma-chrome-launcher": "2.2.0"`

**Comment by alan-agius4** [comment]
This seems like a bug but we'll need to look at a reproduction to find and fix the problem. Can you setup a minimal repro please?

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

**Comment by dadepo** [comment]
As requested here is a repo with code that leads to the error 
https://github.com/dadepo/bug-report-repo

Git cloning, running `npm install`, and then `ng test` should lead to the issue I am seeing.

Ps: I did, and unfortunately that did not resolve the problem :(

**Comment by alan-agius4** [comment]
Hi, 

I was unable to replicate any issue with the repo you submitted.

```
> ngcli-webpack@0.0.0 test /Users/alan/git/cli-repos/bug-report-repo
> ng test

 11% building modules 12/12 modules 0 active26 11 2018 10:07:01.009:INFO [karma-server]: Karma v3.1.1 server started at http://0.0.0.0:9876/
26 11 2018 10:07:01.015:INFO [launcher]: Launching browsers Chrome with concurrency unlimited
26 11 2018 10:07:01.035:INFO [launcher]: Starting browser Chrome
26 11 2018 10:07:06.394:INFO [Chrome 70.0.3538 (Mac OS X 10.14.1)]: Connected on socket CZYn32Wc7ay1EMieAAAA with id 74382277
Chrome 70.0.3538 (Mac OS X 10.14.1): Executed 3 of 3 SUCCESS (0.161 secs / 0.128 secs)
TOTAL: 3 SUCCESS
TOTAL: 3 SUCCESS
Alans-iMac:bug-report-repo alan$ 
```

**Comment by dadepo** [comment]
- I deleted my local copy of the repo
- I cloned the repo afresh and ran `npm install`
- Then I ran `ng test`, and the test still crashes after the test runs.

I made a recording, you can see here https://terminalizer.com/view/16b39bbc28.

Might be an interplay with certain part of the hardware/software I am running? What other troubleshooting would you advice?

**Comment by alan-agius4** [comment]
Hi, do have have installed a non standard pgrep by any chance?

Can you run `which pgrep` in your terminal?

**Comment by dadepo** [comment]
Hey,

Running `which pgrep` outputs: `/usr/local/bin/pgrep`

**Comment by alan-agius4** [comment]
If you have `/usr/local/bin` before `/usr/bin` in your PATH, and a non-standard pgrep installed in `/usr/local/bin`, TreeKill fails to work and spawns hundreds of pgrep processes which typically make the system unusable. 

I suggest you uninstall the custom `pgrep` or change the paths

**Comment by alan-agius4** [comment]
Closing as this is not a bug caused by Angular CLI.

Kindly remember that this issue tracker is not suitable for support requests, please repost your issue on [StackOverflow](http://stackoverflow.com/) using tag `angular-cli`.

If you are wondering why we don't resolve support issues via the issue tracker, please [check out this explanation](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-got-a-question-or-problem).

**Comment by dadepo** [comment]
You have been of tremendous help. The custom pgrep was indeed the problem and uninstalling it, fixes the issue.

> please repost your issue on StackOverflow using tag angular-cli

Oh, I actually did post this as a [question on StackOverflow](https://stackoverflow.com/questions/53419746/how-to-fix-crashing-ng-tests), but it was closed because, well, it was off-topic for StackOverflow...the irony! \o/

Anyways, thanks for your time/help.
